File: app/src/main/java/io/github/hyuwah/muslimcompanionapp/Contract/AyahFetcherContract.java
package io.github.hyuwah.muslimcompanionapp.Contract;

import io.github.hyuwah.muslimcompanionapp.Model.Entity.Ayah;

public interface AyahFetcherContract {


  interface View {

    void fetchSuccess();

    void fetchFailed();

    void showLoading();

    void showResult(Ayah.Data ayah);

    void toggleShareButton(boolean toggle);

    void toggleFavButton(boolean toggle);

  }


  interface Presenter {

    void fetchRandomAyah();

    void fetchAyah(int id);

    void setAyahEdition(String edition);

  }



}

File: app/src/main/java/io/github/hyuwah/muslimcompanionapp/Contract/AyahFetcherContract.kt
package io.github.hyuwah.muslimcompanionapp.Contract

import io.github.hyuwah.muslimcompanionapp.Model.Entity.Ayah

interface AyahFetcherContract {
    interface View {
        fun fetchSuccess()
        fun fetchFailed()
        fun showLoading()
        fun showResult(ayah: Ayah.Data)
        fun toggleShareButton(toggle: Boolean)
        fun toggleFavButton(toggle: Boolean)
    }

    interface Presenter {
        fun fetchRandomAyah()
        fun fetchAyah(id: Int)
        fun setAyahEdition(edition: String)
    }
}